File: source_funcs/go_transcoder/SUM_MATRIX_ELEMENT_ELEMENT_INTEGER_DIVISION_ROW_COLUMN_1/SUM_MATRIX_ELEMENT_ELEMENT_INTEGER_DIVISION_ROW_COLUMN_1.go
package main

func f_gold(n int) int {
	var (
		ans  int = 0
		temp int = 0
		num  int
	)
	for i := int(1); i <= n && temp < n; i++ {
		temp = i - 1
		num = 1
		for temp < n {
			if temp+i <= n {
				ans += i * num
			} else {
				ans += (n - temp) * num
			}
			temp += i
			num++
		}
	}
	return ans
}

File: rust_funcs/cpp_transcoder/SUM_MATRIX_ELEMENT_ELEMENT_INTEGER_DIVISION_ROW_COLUMN_1/SUM_MATRIX_ELEMENT_ELEMENT_INTEGER_DIVISION_ROW_COLUMN_1.rs
fn min(x: i32, y: i32) -> i32 {if x < y { x } else { y } }
fn max(x: i32, y: i32) -> i32 {if x > y { x } else { y }}
fn cmpfunc(a: &i32, b: &i32) -> std::cmp::Ordering {a.cmp(b)}
fn len(arr: &[i32]) -> usize {arr.len()}
fn sort(arr: &mut [i32]) {arr.sort_by(cmpfunc);}


use std::f32;

fn f_gold(n: i32) -> i32 {
    let mut ans = 0;
    let mut temp = 0; 
    let mut num;

    for i in 1..=n {
        if i <= n && temp < n {
            num = 1;
            temp = i - 1;
            while temp < n {
                if temp + i <= n {
                    ans += i * num;
                } else {
                    ans += (n - temp) * num; 
                }
                temp += i;
                num += 1;
            }
        }
    }
    ans
}
